### PR TITLE
ENH: Switch Github Actions macOS environment

### PR DESCRIPTION
**Pull request by jhlegarreta**
Switch Github Actions macOS environment version to `macos-11`.

`macOS-10.15` is being deprecated and support will end by 8/30/2022:
https://github.com/actions/virtual-environments/issues/5583

Related to recent warnings in the Azure Pipelines macOS environment
being used in the main ITK repository:
https://github.com/InsightSoftwareConsortium/ITK/pull/3532/commits/e5595dee1917e94ecc3c7f94873841b0f522d3c0

### CODE REVIEW

**Comment by jhlegarreta** [comment]
Ubuntu Superbuild failures are unrelated:
https://github.com/InsightSoftwareConsortium/ITKSphinxExamples/runs/7705570903?check_suite_focus=true#step:8:21856

**Review by dzenanz — APPROVED** [review]

**Comment by dzenanz** [comment]
Should we wait until that is fixed before merging this?

**Comment by jhlegarreta** [comment]
> Should we wait until that is fixed before merging this?

I do not think so. Although merging is likely to render the CI badge as failing, the macOS-10.15 environment will anyways be unsupported by the end of the month.

**Comment by dzenanz** [comment]
@tbirdso is the fix expected before the end of the month?

**Comment by tbirdso** [comment]
@dzenanz I am not currently clear on the roadmap for addressing [https://github.com/InsightSoftwareConsortium/ITK/issues/3528](https://github.com/InsightSoftwareConsortium/ITK/issues/3528). I have pinged maintainers in that issue in regards to how it may impact the v5.3 roadmap.

**Comment by thewtex** [comment]
Thanks for this update, @jhlegarreta !


> 142: AttributeError: PyCapsule_Import "_ITKCommonPython._C_API" is not valid

@tbirdso addressed this via https://github.com/InsightSoftwareConsortium/ITKMeshToPolyData/pull/50


https://github.com/InsightSoftwareConsortium/ITKSphinxExamples/pull/392 bumps to ITK 5.3 RC 4 -- a rebased after that is in will hopefully be green.

**Comment by dzenanz** [comment]
It looks like the checks are not running?

**Comment by jhlegarreta** [comment]
Re-triggered manually :
https://github.com/InsightSoftwareConsortium/ITKSphinxExamples/actions/runs/2876023121

**Comment by jhlegarreta** [comment]
Push-forced. Re-rerunning properly now.

**Pull request merged**
